### DISCUSSION TITLE
JIT whitelist

### DISCUSSION THREAD

**Discussion by jenstornell** [tailwindlabs/tailwindcss#9308] [Help] [answered]
## The problem

With the old Purge we could whitelist classes that Tailwind could not fetch from the code. It could happen when generating classes dynamically, for example in Vue files.

The recommendation is to write out full class names but sometimes it's more practical to combine strings for it. When creating Tailwind color tools, it's especially helpful.

```js
let shade = 500;
let color = "green";
let html = '<div class="`bg-${color}-${shade}`"></div>';
```

JIT is at least 10 times better but I miss the choice to whitelist classes.

## Solution 1

Similar to how it worked with Purge, we could have a whitelist in the config file.

## Solution 2

Instead of having the whitelist in the config-file which could be bloated, we could have it in an own file like `tailwind.whitelist.js`.

**Comment by wongjn** [comment, marked as answer]
Tailwind v3 still has the safelisting configuration, [see the documentation](https://tailwindcss.com/docs/content-configuration#safelisting-classes).

Also, since the `tailwind.config.js` file is just a normal Common JS file, you can build/import your safelist however you like, for example:

`tailwind.safelist.js`:
```js
module.exports = [
  'text-center',
  'text-left',
  { pattern: /^bg-(red|green|blue)-(100|200|300)$/ },
  // …
];
```

`tailwind.config.js`:
```js
const tailwindSafelist = require('./tailwind.safelist');

module.exports = {
  safelist: tailwindSafelist,
  theme: {
    extend: {
      // …
    },
  },
};
```

**Reply by jenstornell** [reply]
Yes, that works. Not sure why I missed that. Probably read some old docs. Nice solution to keep the safelist in an external file.
